OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -62,7 +62,28 @@ public class ArmamentInfo : UpgradableTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		WeaponInfo weaponInfo = null;
+
+		// Cache initial modified range using sentinal value to mean not cached yet
+		WDist modifiedRange = WDist.MaxValue;
+
 		public override object Create(ActorInitializer init) { return new Armament(init.Self, this); }
+
+		public WeaponInfo GetWeaponInfo(World w)
```

Comment:
It would be cleaner to declare `Lazy<WeaponInfo> weaponInfo` above, and then expose a `public WeaponInfo Weapon { get { weaponInfo.Value; } }`.